Xamarin.Forms.Platform.iOS/Platform.cs

```diff
@@ -462,6 +463,9 @@ static void PresentPopUp(UIWindow window, UIAlertController alert, ActionSheetAr
 
 			var wrapper = new ModalWrapper(modalRenderer);
 
+			if (Device.Idiom == TargetIdiom.Tablet && modal.OnThisPlatform().ModalPresentationStyle() == PlatformConfiguration.iOSSpecific.UIModalPresentationStyle.FormSheet)
```

Comment:
Thoughts on moving this check into the constructor with something like this

```c#
var elementConfiguration = modal.Element as IElementConfiguration<Page>;
			if(elementConfiguration?.On<PlatformConfiguration.iOS>().ModalPresentationStyle() == PlatformConfiguration.iOSSpecific.UIModalPresentationStyle.FormSheet)
				ModalPresentationStyle = UIKit.UIModalPresentationStyle.FormSheet;
```

I'm thinking having this aspect of the setup not being a burden on the caller might be more reliable down the road. Thoughts?

Reply:
I see your point and might be a nice optimization, but I don't see where you'd want to put it/ Constructor of `Platform`?

Reply:
My thinking was to just put it inside the *ModalWrapper* constructor.

Reply:
Ah of course, wow. Overlooked that. That's what you get for coding over the weekend. Absolutely right on moving it to there and not have an additional check when the actual modal is shown. Will update this in a second.